### ISSUE TITLE
DTZ values are incorrect for KPvK and KvKP

### ISSUE THREAD

**Issue by GabrielPereyra** [niklasf/python-chess#281]
From my understanding of DTZ values, they are half-moves until mate, with the sign indicating the player mating with respect to the side to play. However, for a number of fens, `probe_dtz` returns values that differ significantly from stockfish's evaluation (taking into account that stockfish returns moves to mate).

```python
tablebases = chess.syzygy.open_tablebases('...')
fen = '8/8/2P5/7k/3K4/8/8/8 w - - 0 64'
board = chess.Board(fen)
print tablebases.probe_dtz(board) # returns 1 (but stockfish says this is a mate in 10).
```

If I run stockfish, it finds that this position is a mate in 10. This happens for a large number of KPvK boards that I have checked. For other syzygy endgames (KRvK, KQvK, etc), the DTZ values seem correct and agree with stockfish.

Any ideas what could be causing these values to be off?

version==0.22.1

**Comment by niklasf** [comment]
DTZ values are half-moves until the next **capture, or pawn move, or to mate** while keeping the optimal outcome under the 50-move rule. Captures or pawn moves guarantee that progress is beeing made. (There is only a limited number of them.)

DTZ is in some ways superior to DTM: Imagine a position where you already played 99 half-moves since the last capture or pawn move. Say you can either capture a piece or deliver mate in 2. Then DTM would be 2, but following the DTM line allows your opponent to claim a draw under the 50-move rule. If the capture leads to a winning position DTZ would be 1, and capturing the piece to reset the 50-move counter would indeed be optimal play.

On the other hand DTZ-optimal play is sometimes a bit unnatural. For example here the useless Qg5+ forces a capture as soon as possible while still keeping the win in hand: https://syzygy-tables.info/?fen=8/8/8/7k/K7/R7/3Q4/8_w_-_-_0_1

tl;dr: This is by design.
